raft/log.go

```diff
@@ -181,9 +181,13 @@ func (l *raftLog) unstableEntries() []pb.Entry {
 // If applied is smaller than the index of snapshot, it returns all committed
 // entries after the index of snapshot.
 func (l *raftLog) nextCommittedEnts() (ents []pb.Entry) {
-	off := max(l.applied+1, l.firstIndex())
-	if l.committed+1 > off {
-		ents, err := l.slice(off, l.committed+1, l.maxNextCommittedEntsSize)
+	if l.hasPendingSnapshot() {
+		// See comment in hasNextCommittedEnts.
```

Comment:
You could even structure this method as

1. call hasNextCommittedEnts; if false, return early.
2. otherwise, do the slice.

That way there's no duplication in logic between the two and one need not worry about them diverging via a future change.

Reply:
+1